os2mo_tools/mo_api.py

```diff
@@ -102,25 +102,49 @@ class Employee(MOData):
     def __init__(self, uuid, mo_url=DEFAULT_MO_URL):
         """Initialize the employee by specifying the URL prefix."""
         super().__init__(uuid)
-        self.url = mo_url + '/e/' + self.uuid
+        self.url = mo_url + "/e/" + self.uuid
 
 
 def get_ous(org_id=ORG_ROOT, mo_url=DEFAULT_MO_URL):
     """Get all organization units belonging to org_id."""
-    return mo_get(mo_url + '/o/' + org_id + '/ou/')['items']
+    result = []
+    ou_url = mo_url + "/o/" + org_id + "/ou/"
+    total_ous = mo_get("{}?limit=1".format(ou_url))["total"]
+    offset = 1000
+    start = 0
+
+    while len(result) < total_ous and start < total_ous:
+        result += mo_get(
+            "{}?limit={}&start={}".format(ou_url, offset, start)
+        )["items"]
+        start += offset
+
+    return result
 
 
 def get_employees(org_id=ORG_ROOT, mo_url=DEFAULT_MO_URL):
     """Get all employees belonging to the given organization."""
-    return mo_get(mo_url + '/o/' + org_id + '/e/')['items']
+    result = []
+    employee_url = mo_url + "/o/" + org_id + "/e/"
+    total_employees = mo_get("{}?limit=1".format(employee_url))["total"]
```

Comment:
You're doing two requests: One to get the total, and another to get the items — possibly followed by more to fetch the remainder. Why not just check `len(items) + offset >= total` after a fetch? Also, wouldn't this be rather nice as a generator using `yield from`?

```json
{
  "items": […],
  "offset": 100,
  "total": 94
}
```

Reply:
I tend to disagree. I find the initial call for getting the count to be a nice way of handling this pagination thing. Will try to come up with a generator, though.